### PR TITLE
Exposing engine extra info in block RPC

### PR DESCRIPTION
**Pull request by tomusdrw**
Addresses: #3148

### CODE REVIEW

File: rpc/src/v1/types/block.rs [hunk 1]
```diff
@@ -130,12 +135,16 @@ mod tests {
...
...0x0000000000000000000000000000000000000000","miner":"0x0000000000000000000000000000000000000000","stateRoot":"0x0000000000000000000000000000000000000000000000000000000000000000","transactionsRoot":"0x0000000000000000000000000000000000000000000000000000000000000000","receiptsRoot":"0x0000000000000000000000000000000000000000000000000000000000000000","number":"0x0","gasUsed":"0x0","gasLimit":"0x0","extraData":"0x","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","timestamp":"0x0","difficulty":"0x0","totalDifficulty":"0x0","sealFields":["0x","0x"],"extraInfo":{"mixHash":"0x0000000000000000000000000000000000000000000000000000000000000000","nonce":"0x0000000000000000"},"uncles":[],"transactions":[],"size":"0x45"}"#);
```
Comment (gavofyork) [inline rpc/src/v1/types/block.rs]: as i pointed out in the issue, the point of `extra_info` is that the **keys are placed directly into the block/header JSON object**. this is required in order to be compatible with the existing geth behaviour.

**Comment by tomusdrw** [comment]
Fixed. Sorry for the issues, I've misunderstood the solution you've proposed.

**Comment by gavofyork** [comment]
all good.

**Comment by stevenroose** [comment]
Great! When will this land in the beta-release Docker image? :)

**Pull request merged**